OpenGL/Constructs/Geometry.cs

```diff
@@ -52,28 +33,62 @@ public static Vector3[] CalculateNormals(Vector3[] vertexData, int[] elementData
         /// <returns></returns>
         public static Vector3[] CalculateNormals(Vector3[] vertexData, uint[] elementData)
         {
-            Vector3 b1, b2, normal;
+            return CalculateNormals(vertexData.AsSpan(), elementData.AsSpan());
```

Comment:
This method now has the ability to throw an exception, whereas before it wouldn't unless the code threw an `IndexOutOfRangeException` during calculation (which it can still do if an elementData value is incorrect I suppose).  I think the new behaviour is actually better, but I always like to think about how this might impact users...  Is there a potential use case of CalculateNormals that was somehow useful in the past with a non multiple of 3 `elementData` array?  This is more of a thought experiment to make sure it is brought up, and not really a request for change.